.github/workflows/gh-ci.yaml

```diff
@@ -0,0 +1,189 @@
+name: GH_CI
+on:
+  push:
+    branches:
+      - develop
+      - master
+  pull_request:
+    branches:
+      - develop
+      - master
+
+defaults:
+  run:
+    shell: bash -l {0}
+
+jobs:
+  main_tests:
+    if: "github.repository == 'MDAnalysis/mdanalysis'"
+    runs-on: ${{ matrix.os }}
+    strategy:
+        matrix:
+          os: [ubuntu-latest, ]
+          python-version: [3.6, 3.7, 3.8]
+          run_type: [full, ]
+          install_hole: [True, ]
+          codecov: [True, ]
+          include:
+            - name: macOS
+              os: macOS-latest
+              python-version: 3.6
+              run_type: full
+              install_hole: True
+              codecov: True
+            - name: minimal-ubuntu
+              os: ubuntu-latest
+              python-version: 3.6
+              run_type: min
+              install_hole: False
+              codecov: True
+            - name: asv_check
+              os: ubuntu-latest
+              python-version: 3.6
+              run_type: full
+              install_hole: False
+              codecov: False
+    env:
+      CYTHON_TRACE_NOGIL: 1
+      MPLBACKEND: agg
+      MACOSX_DEPLOYMENT_TARGET: 10.9
+      GH_OS: ${{ matrix.os }}
+
+    steps:
+    - uses: actions/checkout@v2
+
+    - name: setup_osx
+      if: startsWith(matrix.os, 'macOS')
+      run: |
+        echo "OS_NAME=osx" >> $GITHUB_ENV
+        ulimit -S -n 2048
+
+    - name: setup_linux
+      if: startsWith(matrix.os, 'ubuntu')
+      run: |
+        echo "OS_NAME=linux" >> $GITHUB_ENV
+
+    - name: setup_miniconda
+      uses: conda-incubator/setup-miniconda@v2
+      with:
+        python-version: ${{ matrix.python-version }}
+        auto-update-conda: true
+        channel-priority: flexible
+        # channel clash even though you have the same definition..
+        #channels: biobuilds, conda-forge
+        add-pip-as-python-dependency: true
+        # mamba does not currently support pip in env files
+        # to enable post gh/mamba-org/mamba/issues/589
+        #mamba-version: "*"
+        architecture: x64
+        environment-file: maintainer/test-envs/conda-${{ matrix.run_type }}.yaml
+        activate-environment: mda-tests
+    
+    - name: check_setup
+      run: |
+        # Check OS
+        echo "OS: ${OS_NAME}"
+        # Check python
+        which python
+        which pip
+        conda info
+        # TODO: adequate tqdm installed, setup.py still overrides...
+        conda list
+
+    - name: install_asv
+      if: contains(matrix.name, 'asv_check')
+      run: |
+        pip install asv "setuptools<45.0.0"
+
+    - name: install_hole
+      if : matrix.install_hole
+      run: |
+        bash ./maintainer/install_hole.sh ${OS_NAME} "${HOME}"
+        HOLE_BINDIR=${HOME}/hole2/exe
```

Comment:
This (and the export below) won't work for HOLE. This can be seen from the lack of coverage for hole2. The issue here, I think, is that env variables and exports aren't shared between job steps by default.

We'll have to do something similar to what we do with setup_linux.